src/npc.cpp

```diff
@@ -1712,7 +1712,15 @@ int npc::follow_distance() const
           g->m.has_flag( TFLAG_GOES_UP, g->u.pos() ) ) ) {
         return 1;
     }
-    // TODO: Allow player to set that
+    // If NPC doesn't see player, change follow distance to 2
+    if( !sees( g->u ) ) {
+        return 2;
+    }
+    // Uses ally_rule follow_distance_close to determine if should follow by 2 or 4 tiles
+    if( !rules.has_flag(
```

Comment:
Please call it follow_distance_4. It's better to rename a variable than have confusing logic.

Also, move this up before the `if ( !sees( g->u ) ) {` test, because this test is a lot faster than the sees test so we should do it first.